typescript/appconfig-hosted-configuration-lambda-extension/index.ts

```diff
@@ -0,0 +1,89 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnApplication, CfnEnvironment, CfnConfigurationProfile, CfnHostedConfigurationVersion, CfnDeploymentStrategy, CfnDeployment } from '@aws-cdk/aws-appconfig';
+import { Function, AssetCode, Runtime, LayerVersion } from '@aws-cdk/aws-lambda';
+import { Effect, PolicyStatement } from '@aws-cdk/aws-iam';
+
+export class AppConfigHostedConfigurationStack extends cdk.Stack {
+	constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+		super(scope, id, props);
+
+		const application = new CfnApplication(this,'AppConfigApplication', {
+			name: 'AppConfigSampleApplication',
+			description: 'Sample AppConfig Application using CDK'
+		});
+
+		const environment = new CfnEnvironment(this, 'LambdaDevelopmentEnvironment', {
+			applicationId: application.ref,
+			name: 'AppConfigSampleLambdaDevelopmentEnvironment',
+			description: 'Sample AppConfig Development environment for Lambda implementation'
+		});
+
+		const configurationProfile = new CfnConfigurationProfile(this, 'ConfigurationProfile', {
+			applicationId: application.ref,
+			name: 'AppConfigSampleConfigurationProfile',
+			locationUri: 'hosted',
+			description: 'Sample AppConfig configuration profile'
+		});
+
+		const hostedConfigurationProfile = new CfnHostedConfigurationVersion(this, 'HostedConfigurationProfile', {
+			applicationId: application.ref,
+			configurationProfileId: configurationProfile.ref,
+			contentType: 'application/json',
+			content: '{\"boolEnableLimitResults\": true, \"intResultLimit\":5}',
+			latestVersionNumber: 1
+		});
+
+		hostedConfigurationProfile.addMetadata('description', 'Sample AppConfig hosted configuration profile content');
+
+		const deploymentStrategy = new CfnDeploymentStrategy(this, 'DeploymentStrategy', {
+			name: 'Custom.AllAtOnce',
+			deploymentDurationInMinutes: 0,
+			growthFactor: 100,
+			finalBakeTimeInMinutes: 0,
+			replicateTo: 'NONE',
+			growthType: 'LINEAR',
+			description: 'Sample AppConfig deployment strategy - All at once deployment (i.e., immediate)'
+		});
+
+		const deployment = new CfnDeployment(this, 'Deployment', {
+			applicationId: application.ref,
+			configurationProfileId: configurationProfile.ref,
+			configurationVersion: '1',
```

Comment:
Another magic number. This should be:
```suggestion
			configurationVersion: constedConfigurationProfile.latestVersionNumber,
```

Reply:
@NGL321 not exactly intuitive, but `hostedConfigurationProfile.ref` should be used instead. `latestVersionNumber` is undefined here.